onnx/defs/controlflow/defs.cc

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Facebook Inc. and Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "onnx/defs/schema.h"
+using namespace onnx;
+
+using SupportType = onnx::OpSchema::SupportType;
+
+std::vector<std::string> alltypes = {"tensor(float)",
+                                     "tensor(int32)",
+                                     "tensor(string)",
+                                     "tensor(bool)",
+                                     "tensor(uint8)",
+                                     "tensor(int8)",
+                                     "tensor(uint16)",
+                                     "tensor(int16)",
+                                     "tensor(int64)",
+                                     "tensor(float16)",
+                                     "tensor(double)"};
+
+OPERATOR_SCHEMA(If)
+    .SetSupportLevel(SupportType::EXPERIMENTAL)
+    .SetDoc("If conditional")
+    .Input(0, "cond", "Condition for the if", "B")
+    .Attr(
+        "then_branch",
+        "Graph to run if condition is true",
+        AttributeProto::GRAPH,
+        GraphProto())
+    .Attr(
+        "else_branch",
+        "Graph to run if condition is false",
+        AttributeProto::GRAPH,
+        GraphProto())
+    .TypeConstraint("B", {"tensor(bool)"}, "Only bool");
+
+OPERATOR_SCHEMA(While)
+    .SetSupportLevel(SupportType::EXPERIMENTAL)
+    .SetDoc(R"DOC(
+While Loop. This loop has multiple termination conditions:
+
+1) Static trip count. Iteration count specified at compile time. Set by setting
+   the static_trip_count attribute (optional)
+2) Dynamic trip count. Iteration count specified at runtime. Set by specifying
+   the input M, and setting the attribute has_trip_count = true (optional)
+3) Loop conditional net. A net that has a single boolean result that specifies
+   whether the loop should continue. Set by specifying the cond attribute.
+
+   The graph body will only be run if all of the above conditions are true. In
+the event that an optional condition is not specified, that condition is treated
+as if it were always true.
+
+Note that it is the responsiblity of the backend to properly schedule this
+operator with respect to the data dependencies of both the condition net
+and the body net.
+)DOC")
+    .Input(
```

Comment:
it needs to be marked as optional

Also it means that it's an always present in the graph description but might be 'empty string'